src/Microsoft.OData.Client/ALinq/DataServiceQueryProvider.cs

```diff
@@ -172,9 +172,23 @@ internal QueryComponents Translate(Expression e)
 
             UriWriter.Translate(this.Context, addTrailingParens, e, out uri, out version);
             ResourceExpression re = e as ResourceExpression;
-            Type lastSegmentType = re.Projection == null ? re.ResourceType : re.Projection.Selector.Parameters[0].Type;
+            QueryableResourceExpression qre = e as QueryableResourceExpression;
+            Type lastSegmentType;
+
+            if (qre?.Apply != null && qre?.Apply.KeySelectorMap.Count > 0)

```

Comment:
isn't having just this `qre?.Apply?.KeySelectorMap.Count > 0` sufficient here?

Reply:
@ElizabethOkerio Fixed